src/tests/resistance/post_handshake/random_bytes.rs

```diff
@@ -0,0 +1,258 @@
+        },
+        synthetic_node::SyntheticNodeBuilder,
+        util::gen_rand_bytes,
+    },
+};
+
+/// Send given bytes directly to the node after the handshake and return the connection status.
+async fn send_bytes_to_the_node(data: Vec<u8>, debug: bool) -> bool {
+    // Spin up a node instance.
+    let target = TempDir::new().expect(ERR_TEMPDIR_NEW);
+    let mut node = Node::builder()
+        .log_to_stdout(debug)
+        .build(target.path())
+        .expect(ERR_NODE_BUILD);
+    node.start().await;
+
+    // Create a synthetic node and disable handshaking.
+    let synthetic_node = SyntheticNodeBuilder::default()
+        .build()
+        .await
+        .expect(ERR_SYNTH_BUILD);
+
+    let net_addr = node.net_addr().expect(ERR_NODE_ADDR);
+
+    // Create a connection without the handshake.
+    synthetic_node
+        .connect(net_addr)
+        .await
+        .expect(ERR_NODE_CONNECT);
+
+    // Send some random data.
+    let random_data_msg = Payload::RawBytes(data);
+    synthetic_node
+        .unicast(net_addr, random_data_msg)
+        .expect(ERR_SYNTH_UNICAST);
+
+    // Give some time to the node to kill our connection.
+    sleep(WAIT_FOR_DISCONNECT).await;
+
+    let is_connected = synthetic_node.is_connected(net_addr);
+
+    // Gracefully shut down the nodes.
+    synthetic_node.shut_down().await;
+    node.stop().expect(ERR_NODE_STOP);
+
+    is_connected
+}
+
+fn gen_tagged_msg_with_random_data(tag: Tag, len: usize) -> Vec<u8> {
+    let mut msg_content_random = BytesMut::from(Tag::get_tag_str(&tag).as_bytes());
+    msg_content_random.put_slice(&gen_rand_bytes(len));
+    msg_content_random.to_vec()
+}
+
+struct TagRandDataTestCfg {
+    tag: Tag,
+    debug_logs: bool,
+    data_len_normal: usize,
+    data_len_huge: usize,
+}
+
+impl TagRandDataTestCfg {
+    fn with_tag(mut self, tag: Tag) -> Self {
+        self.tag = tag;
+        self
+    }
+}
+
+impl Default for TagRandDataTestCfg {
+    fn default() -> Self {
+        Self {
+            tag: Tag::RawBytes,
+            debug_logs: false,
+            data_len_normal: 15,
+            data_len_huge: 1_000_000,
+        }
+    }
+}
+
+async fn send_tagged_rand_data_to_the_node(cfg: TagRandDataTestCfg) {
+    // Run the test with normal data length.
+    let tagged_random_data = gen_tagged_msg_with_random_data(cfg.tag, cfg.data_len_normal);
+    assert!(
+        !send_bytes_to_the_node(tagged_random_data, cfg.debug_logs).await,
+        "the node shouldn't keep the connection alive after receiving random data"
+    );
+
+    // Run the test with huge data length.
+    let tagged_random_data = gen_tagged_msg_with_random_data(cfg.tag, cfg.data_len_huge);
+    assert!(
+        !send_bytes_to_the_node(tagged_random_data, cfg.debug_logs).await,
+        "the node shouldn't keep the connection alive after receiving random data"
+    );
+}
+
+#[tokio::test]
+#[allow(non_snake_case)]
+async fn r003_t1_RANDOM_DATA_send_completely_random_data() {
```

Comment:
nit: we could prolly create the cases for each message type with a simple macro

Reply:
Done